pkg/kubectl/cmd/plugin/plugin.go

```diff
@@ -238,3 +242,18 @@ func isExecutable(fullPath string) (bool, error) {
 
 	return false, nil
 }
+
+// uniquePathsList deduplicates a given slice of strings without
+// sorting or otherwise altering its order in any way.
+func uniquePathsList(paths []string) []string {
+	seen := map[string]bool{}
+	newPaths := []string{}
+	for _, p := range paths {
+		if seen[p] {
+			continue
+		}
+		seen[p] = true
+		newPaths = append(newPaths, p)
```

Comment:
How about accumulating the count of unique paths in the loop and allocate array, newPaths, outside the loop ?

This way there is no need to call append which saves some memory.

Reply:
I don't think that matters that much here. You'd have to iterate over the paths array twice in that case.